### ISSUE TITLE
mso_tenant : idempotency (running playbook second time)

### ISSUE THREAD

**Issue by 4ndersson** [CiscoDevNet/ansible-mso#92]
When running mso_tenant the second time I get the error beneath. I did a temporary workaround by editing line 56 in mso.py due to some error in the cmp function.

```
            except TypeError:
                pass
                # Fall back to exact comparison for lists of dicts
                #if not cmp(value, superset[key]):
                #    return False
```

**Versions**
ansible 2.9.9
MSO: 2.2(4e)

**Errors**

Running the playbook first time... WORKS!

```
PLAY [mso] ******************************************************************************************

TASK [mso : Add a new tenant] ***********************************************************************
changed: [10.145.145.190 -> localhost]
```

Running the playbook second time... DOES NOT WORK!

```
PLAY [mso] *************************************************************************************************************************************************************************************************
META: ran handlers

TASK [mso : Add a new tenant] ******************************************************************************************************************************************************************************
task path: /Users/andersson/proj/vgr/ansible/roles/mso/tasks/add_tenant.yml:2
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: andersson
<localhost> EXEC /bin/sh -c 'echo ~andersson && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/andersson/.ansible/tmp `"&& mkdir /Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638 && echo ansible-tmp-1596787765.3379-19422-117240935043638="` echo /Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638 `" ) && sleep 0'
Using module file /Users/andersson/Library/Python/3.7/lib/python/site-packages/ansible/modules/network/aci/mso_tenant.py
<localhost> PUT /Users/andersson/.ansible/tmp/ansible-local-19418c4_87feb/tmpp4q4468q TO /Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638/AnsiballZ_mso_tenant.py
<localhost> EXEC /bin/sh -c 'chmod u+x /Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638/ /Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638/AnsiballZ_mso_tenant.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/local/opt/python/bin/python3.7 /Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638/AnsiballZ_mso_tenant.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/var/folders/ys/c8j8l1ms5x94nc82klphsk7w0000gn/T/ansible_mso_tenant_payload_kmnu8633/ansible_mso_tenant_payload.zip/ansible/module_utils/network/aci/mso.py", line 54, in issubset
TypeError: unhashable type: 'dict'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638/AnsiballZ_mso_tenant.py", line 102, in <module>
    _ansiballz_main()
  File "/Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638/AnsiballZ_mso_tenant.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638/AnsiballZ_mso_tenant.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible.modules.network.aci.mso_tenant', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/var/folders/ys/c8j8l1ms5x94nc82klphsk7w0000gn/T/ansible_mso_tenant_payload_kmnu8633/ansible_mso_tenant_payload.zip/ansible/modules/network/aci/mso_tenant.py", line 200, in <module>
  File "/var/folders/ys/c8j8l1ms5x94nc82klphsk7w0000gn/T/ansible_mso_tenant_payload_kmnu8633/ansible_mso_tenant_payload.zip/ansible/modules/network/aci/mso_tenant.py", line 185, in main
  File "/var/folders/ys/c8j8l1ms5x94nc82klphsk7w0000gn/T/ansible_mso_tenant_payload_kmnu8633/ansible_mso_tenant_payload.zip/ansible/module_utils/network/aci/mso.py", line 58, in issubset
  File "/var/folders/ys/c8j8l1ms5x94nc82klphsk7w0000gn/T/ansible_mso_tenant_payload_kmnu8633/ansible_mso_tenant_payload.zip/ansible/module_utils/network/aci/mso.py", line 16, in cmp
TypeError: '>' not supported between instances of 'dict' and 'dict'
fatal: [10.145.145.190 -> localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/var/folders/ys/c8j8l1ms5x94nc82klphsk7w0000gn/T/ansible_mso_tenant_payload_kmnu8633/ansible_mso_tenant_payload.zip/ansible/module_utils/network/aci/mso.py\", line 54, in issubset\nTypeError: unhashable type: 'dict'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638/AnsiballZ_mso_tenant.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638/AnsiballZ_mso_tenant.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/Users/andersson/.ansible/tmp/ansible-tmp-1596787765.3379-19422-117240935043638/AnsiballZ_mso_tenant.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible.modules.network.aci.mso_tenant', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/var/folders/ys/c8j8l1ms5x94nc82klphsk7w0000gn/T/ansible_mso_tenant_payload_kmnu8633/ansible_mso_tenant_payload.zip/ansible/modules/network/aci/mso_tenant.py\", line 200, in <module>\n  File \"/var/folders/ys/c8j8l1ms5x94nc82klphsk7w0000gn/T/ansible_mso_tenant_payload_kmnu8633/ansible_mso_tenant_payload.zip/ansible/modules/network/aci/mso_tenant.py\", line 185, in main\n  File \"/var/folders/ys/c8j8l1ms5x94nc82klphsk7w0000gn/T/ansible_mso_tenant_payload_kmnu8633/ansible_mso_tenant_payload.zip/ansible/module_utils/network/aci/mso.py\", line 58, in issubset\n  File \"/var/folders/ys/c8j8l1ms5x94nc82klphsk7w0000gn/T/ansible_mso_tenant_payload_kmnu8633/ansible_mso_tenant_payload.zip/ansible/module_utils/network/aci/mso.py\", line 16, in cmp\nTypeError: '>' not supported between instances of 'dict' and 'dict'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

PLAY RECAP *************************************************************************************************************************************************************************************************
10.145.145.190             : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

**CONFIG**

```
---
- name: Add a new tenant
  mso_tenant:
    host: "{{ inventory_hostname }}"
    username: "{{ user }}"
    password: "{{ pass }}"
    validate_certs: no
    tenant: CloudTenant
    display_name: CloudTenant
    sites: vbdc
    state: present
  delegate_to: localhost
```

**Comment by xinyuezhao** [comment]
@4ndersson 

Hi, would you mind use 'ansible-galaxy' to upgrade to the latest version of our module, in that version the issued should be fixed.

here is one way to upgrade to the newly version:

Go to: https://github.com/CiscoDevNet/ansible-mso/actions
Select the latest CI build
Under Artifacts download collections and unzip it on Terminal
Get the tar.gz file
Install using ansible-galaxy collection install
Example: ansible-galaxy collection install cisco-mso-1.0.0.tar.gz —force

**Comment by xinyuezhao** [comment]
a small correction, this issue should be fixed in our resealed version 1.0.0

you can directly download the latest official version from [https://galaxy.ansible.com/cisco/mso](url) and then use "ansible-galaxy collection install cisco-mso-1.0.0.tar.gz —force" to upgrade to this version.

**Comment by 4ndersson** [comment]
@xinyuezhao Yes, the ansible-galaxy collection version works much better, both regarding this issue and other modules.
